### PR TITLE
Added user animal type table

### PR DESCRIPTION
**Pull request by jonathankim345**
## Notion ticket link

[User Pet Related Relationships](https://www.notion.so/uwblueprintexecs/User-Pet-Related-Relationships-02bb744573714e9a97dbb08c6bc27a8a?pvs=4)

## Implementation description
* Build out relationship tables for User_AnimalTypes by creating a many-to-many relationship between the User and AnimalType models

## Steps to test
1. Insert into user animal types table when animal types table is empty
Screenshot 2024-07-23 205938
Screenshot 2024-07-23 210536

2. Insert into user animal types table when users types table is empty
Screenshot 2024-07-23 210000
Screenshot 2024-07-23 210434

## What should reviewers focus on?
* The same animal type can have many users
* The same user can have many animal types
Screenshot 2024-07-23 211301

Screenshot 2024-07-23 211210

## Checklist
- [✅] My PR name is descriptive and in imperative tense
- [✅ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [✅] I have run the appropriate linter(s)
- [✅ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/models/index.ts [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
-/* eslint-disable-next-line import/prefer-default-export */
-export const sequelize = new Sequelize(DATABASE_URL, {
+const sequelize = new Sequelize(DATABASE_URL, {
   models: [path.join(__dirname, "/*.model.ts")],
 });
+
+/* eslint-disable-next-line import/prefer-default-export */
+sequelize.addModels([User, AnimalType, UserAnimalTypes]);
+
+User.belongsToMany(AnimalType, {
```
Comment (aanxniee) [inline backend/typescript/models/index.ts:20]: is there a reason why we need to change this file?

Comment (jonathankim345) [reply backend/typescript/models/index.ts:20]: @aanxniee Yes unfortunately, I was receiving a circular dependency error. I spoke with Sayi and he told me that while the other people on the team had one, they weren't modelling a many to many relationship like in this PR. I tried to resolve them in their specific tables but it didn't work. The information I found online suggested having some kind of outside file to resolve all the relationships like an index file. I can try and find other solutions, I didn't want to edit the index file either

Comment (aanxniee) [reply backend/typescript/models/index.ts:20]: hmm i see @laks0407 @jerry-cheng5

**Comment by jerry-cheng5** [comment]
closing PR - opened a new ticket to finish this

**Pull request closed without merging**